validphys2/src/validphys/loader.py

```diff
@@ -62,46 +62,69 @@ class CutsNotFound(LoadFailedError): pass
 
 class PDFNotFound(LoadFailedError): pass
 
+class ProfileNotFound(LoadFailedError): pass
+
 class RemoteLoaderError(LoaderError): pass
 
 class InconsistentMetaDataError(LoaderError): pass
 
+def _get_nnpdf_profile(profile_path=None):
+    """Returns the NNPDF profile as a dictionary
+    If no ``profile_path`` is provided it will be autodiscovered in the following order:
+
+    Environment variable $NNPDF_PROFILE_PATH
+    {sys.prefix}/share/NNPDF/nnprofile.yaml
+    {sys.base_prefix}/share/NNPDF/nnprofile.yaml
+
+    If no profile is found a LoaderError will be thrown
+    """
+    profile_path = os.environ.get("NNPDF_PROFILE_PATH", profile_path)
+    if profile_path is None:
+        # Check both sys paths
+        prefix_paths = [sys.prefix, sys.base_prefix]
+        for prefix in prefix_paths:
+            check = pathlib.Path(prefix) / "share/NNPDF/nnprofile.yaml"
+            if check.exists():
+                profile_path = check
+                break
+    if profile_path is None:
+        raise LoaderError("Missing an NNPDF profile file")
+
+    mpath = pathlib.Path(profile_path)
+    try:
+        profile_dict = yaml.safe_load(mpath.open("r"))
```

Comment:
It is generally recommended to use `open` as a context manager.
```suggestion
        with mpath.open() as f:
            profile_dict = yaml.safe_load(f)
```